retrofit/src/main/java/retrofit/converter/GsonConverter.java

```diff
@@ -75,9 +75,10 @@ public GsonConverter(Gson gson, String charset) {
     }
   }
 
-  @Override public TypedOutput toBody(Object object) {
+  @Override public TypedOutput toBody(Object object, Type type) {
     try {
-      return new JsonTypedOutput(gson.toJson(object).getBytes(charset), charset);
+      String json = gson.toJson(object, type);
```

Comment:
Slight potential for a behavior change here. For example, if `Type` is an interface, Gson may very well freak out attempting to find a converter.

Regardless, I think this is the right behavior.